staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -144,15 +144,7 @@ func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, re
 		// The filter doesn't filter out any object.
 		wc.internalPred = storage.Everything
 	}
-
-	// The etcd server waits until it cannot find a leader for 3 election
-	// timeouts to cancel existing streams. 3 is currently a hard coded
-	// constant. The election timeout defaults to 1000ms. If the cluster is
-	// healthy, when the leader is stopped, the leadership transfer should be
-	// smooth. (leader transfers its leadership before stopping). If leader is
-	// hard killed, other servers will take an election timeout to realize
-	// leader lost and start campaign.
-	wc.ctx, wc.cancel = context.WithCancel(clientv3.WithRequireLeader(ctx))
+	wc.ctx, wc.cancel = context.WithCancel(ctx)
```

Comment:
Why did you change this?

[Also, if we want to change that, it seems like a separate PR to me]

Reply:
Or you rather wanted to use watchContext here instead?

Reply:
OK - never mind, it seems that you moved this around.

So let's (as I mentioned above) move those cleanups/refactors to a separate PR.

Reply:
Needed to change that to make sure that calls to etcd client `Watch` and `RequestProgressNotify` have the same context metadata. This is not a useful otherwise.